src/sentry/eventstore/models.py

```diff
@@ -736,6 +739,14 @@ def from_event(cls, event: Event, group: Group) -> GroupEvent:
 
     @property
     def occurrence(self) -> IssueOccurrence:
+        if not self._occurrence and self.occurrence_id:
+            self._occurrence = IssueOccurrence.fetch(self.occurrence_id, self.project_id)
+            if self._occurrence is None:
+                logger.error(
+                    "Failed to fetch occurrence for event",
+                    extra={"group_id": self.group_id, "occurrence_id": self.occurrence_id},
+                )
```

Comment:
do we want to return something else here? feel like this should/could fail `typing`

Reply:
Yeah... I'll update the type hint here

Reply:
I think that ideally, in the future this should probably raise if we can't find the occurrence, since there shouldn't be a `GroupEvent` without an occurrence. That'll have to wait until after we migrate perf issues across though.